src/vim-loader/loader.ts

```diff
@@ -25,23 +25,82 @@ export class Loader {
   }
 
   async load (bfast: BFast, settings: VimConfig) {
-    let document: IDocument | undefined
-
     const mode = settings.download
     if (mode === 'download') await bfast.forceDownload()
 
-    await Document.createFromBfast(bfast, mode === 'stream').then(
-      (d) => (document = d)
-    )
-    if (!document) {
-      throw Error('Could not load parse document.')
-    }
+    let g3d: G3d | undefined
+    let strings: string[] | undefined
+
+    let instanceToElement: number[] | undefined
+    let elementIds: number[] | undefined
+
+    const doc = await VimDocument.createFromBfast(bfast)
+
+    await Promise.all([
+      Loader.requestG3d(bfast, mode === 'stream').then((g) => (g3d = g)),
+      Loader.requestStrings(bfast).then((s) => strings = s),
+      doc.node.getAllElementIndex().then(
+        (array) => (instanceToElement = array)
+      ),
+      doc.element.getAllId().then((array) => (elementIds = array))
+    ])
 
-    const scene = document.g3d
-      ? this.sceneBuilder.createFromG3d(document!.g3d, settings.transparency)
+    const scene = g3d
+      ? this.sceneBuilder.createFromG3d(g3d, settings.transparency)
       : new Scene(this.sceneBuilder)
 
-    const vim = new Vim(document!, scene, settings)
+    const elementToInstance = Loader.invert(instanceToElement!)
```

Comment:
These inverts would be better in the constructor. So the constructor would only have 2 arguments.

Reply:
and the invert method could be removed from here too.